daml-lf/kv-support/src/main/scala/com/daml/lf/kv/contracts/ContractConversions.scala

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf.kv.contracts
+
+import com.daml.lf.transaction.{TransactionCoder, TransactionOuterClass}
+import com.daml.lf.value.{Value, ValueCoder}
+
+object ContractConversions {
+
+  def encodeContractInstance(
+      coinst: Value.VersionedContractInstance
+  ): Either[ValueCoder.EncodeError, RawContractInstance] =
+    TransactionCoder
+      .encodeContractInstance(ValueCoder.CidEncoder, coinst)
+      .map(contractInstance => RawContractInstance(contractInstance.toByteString))
+
+  def decodeContractInstance(
+      rawContractInstance: RawContractInstance
+  ): Either[ValueCoder.DecodeError, Value.VersionedContractInstance] = {
+    val contractInstance =
```

Comment:
this can throw I assume? Seems a bit odd to have a method that returns an `Either` but only have it handle half of the errors in the method.

Reply:
Not sure if I get this one, what do you mean by handling half of the errors? I'm not handling errors here.

Reply:
Maybe handling was the wrong word. What I meant is that errors from `parseFrom` are turned into an exception while errors from `decodeVersionedContractInstance` are turned into a `Left` that seems like a somewhat confusing API and probably also unhelpful since I think both of them are internal errors for you (meaning you should never hit either of them).

Reply:
Good point, I will think about unifying it.